src/factories/beacon/InverterBeacon.sol

```diff
@@ -0,0 +1,150 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity 0.8.23;
+
+// External Dependencies
+import {Ownable2Step} from "@oz/access/Ownable2Step.sol";
+import {ERC165} from "@oz/utils/introspection/ERC165.sol";
+import {Ownable} from "@oz/access/Ownable.sol";
+
+// External Interfaces
+import {IBeacon} from "@oz/proxy/beacon/IBeacon.sol";
+
+//Internal Interfaces
+import {IInverterBeacon} from "src/factories/beacon/IInverterBeacon.sol";
+/**
+ * @title Beacon
+ */
+
+contract InverterBeacon is IInverterBeacon, ERC165, Ownable2Step {
+    //--------------------------------------------------------------------------------
+    // ERC-165 Public View Functions
+
+    /// @inheritdoc ERC165
+    function supportsInterface(bytes4 interfaceId)
+        public
+        view
+        virtual
+        override
+        returns (bool)
+    {
+        return interfaceId == type(IInverterBeacon).interfaceId
+            || interfaceId == type(IBeacon).interfaceId
+            || super.supportsInterface(interfaceId);
+    }
+
+    //--------------------------------------------------------------------------------
+    // State
+
+    /// @dev The beacon's implementation address.
+    address private _implementation;
+
+    /// @dev The beacon's current implementation pointer.
+    address private _currentImplementation;
+
+    /// @dev Is the beacon shut down / in emergency mode
+    bool public _emergencyMode;
```

Comment:
Does this one have to be `public`?

Reply:
Yeah youre right. I had it intially public without the underscore but changed my mind later